File: union/src/main/java/com/union/app/messaging/service/EventService.java
package com.union.app.messaging.service;

import com.union.app.user.model.User;
import com.union.app.messaging.dto.EventCreateDto;
import com.union.app.messaging.dto.EventUpdateDto;
import com.union.app.messaging.model.Event;
import java.util.List;

public interface EventService {
    Event createEvent(Long unionId, EventCreateDto dto, User creator);

    Event updateEvent(Long eventId, EventUpdateDto dto, User currentUser);

    void deleteEvent(Long eventId, User currentUser);

    List<Event> getEventsForUnion(Long unionId);
}

File: union/frontend/src/services/event.service.ts
import api from './api';
import { Event, EventCreateDto, EventUpdateDto } from '../types';

export const eventService = {
  createEvent: async (unionId: number, data: EventCreateDto): Promise<Event> => {
    const response = await api.post(`/events/createEvent/${unionId}`, data);
    return response.data;
  },

  updateEvent: async (eventId: number, data: EventUpdateDto): Promise<Event> => {
    const response = await api.put(`/events/updateEvent/${eventId}`, data);
    return response.data;
  },

  deleteEvent: async (eventId: number): Promise<void> => {
    await api.delete(`/events/deleteEvent/${eventId}`);
  },

  getEventsByUnion: async (unionId: number): Promise<Event[]> => {
    const response = await api.get(`/events/listEvents/${unionId}`);
    return response.data;
  },
};
